caches/lru_variants.cpp

```diff
@@ -551,4 +551,54 @@ void S4LRUCache::evict()
     segments[0].evict();
 }
 
+/*
+  Random: Random eviction
+*/
+bool RandomCache::lookup(SimpleRequest* req)
+{
+    CacheObject obj(req);
+    auto it = _cacheSet.find(obj);
+    if (it != _cacheSet.end()) {
+        // log hit
+        LOG("h", 0, obj.id, obj.size);
+        return true;
+    }
+    return false;
+}
 
+void RandomCache::admit(SimpleRequest* req)
+{
+    const uint64_t size = req->getSize();
+    // object feasible to store?
+    if (size > _cacheSize) {
+        LOG("L", _cacheSize, req->getId(), size);
+        return;
+    }
+    // check eviction needed
+    while (_currentSize + size > _cacheSize) {
+        evict();
+    }
+    // admit new object
+    CacheObject obj(req);
+    _cacheVector.push_back(obj);
+    _cacheSet.insert(obj);
+    _currentSize += size;
+    LOG("a", _currentSize, obj.id, obj.size);
+}
+
+void RandomCache::evict(SimpleRequest* req)
+{
+}
+
```

Comment:
This function is for workload that include invalidation requests. I know this is not implemented in the the main method, but some people are using it. Would you mind implementing the eviction of a specific object id?

This is a rare operation and it is ok to implement it in O(n) for the vector scan, if you'd like to continue to use an unordered_set instead of an unordered_map.

Reply:
That should do the trick. I didn't test it, though.